### PR TITLE
[model, ci] fix: Wan Ulysses SP sync-path attention correctness

### PR DESCRIPTION
**Pull request by OnePunchMonk**
### What does this PR do?

Fixes the two gaps flagged in #1140 (opened as a follow-up from #1139's review):

1. **`attention_mask` was a no-op for `flash_attention_3`/`sageattention`.** `wrapped_flash_attention_3` and `wrapped_sageattention` in `veomni/models/transformers/wan/modeling_wan.py` accepted `attention_mask` only via `**kwargs` and never read it, so the Ulysses SP tail-padding mask built in `WanModel.forward` had no effect for either backend (only `eager_attention_forward` honored it). Added `_trim_kv_tail_padding`, which drops the padded K/V rows the mask marks before the kernel call (neither kernel takes a mask argument of its own) and re-pads the output back to the original length afterward.

2. **The sync self-attention path never did the Ulysses all-to-all.** `SelfAttention.forward`'s sync path (`sp_async=False`, the only path Wan currently uses -- nothing in the repo sets `sp_async=True`) never redistributed Q/K/V via all-to-all before calling into `AttentionModule`; `gather_seq_scatter_heads_qkv` was defined in this file but never called. Every rank's backend attention therefore ran only over its own local SP shard as both query and key/value, instead of the true full sequence with a subset of heads -- so Ulysses SP was not numerically equivalent to non-SP attention for Wan on any of the three backends. Added the missing `gather_seq_scatter_heads_qkv` / `gather_heads_scatter_seq` pair around the attention call, matching the pattern already used in `flux/modeling_flux.py`. The global tail-padding mask no longer needs the sync path's old per-rank slicing, since both paths now see the true full sequence.

### Checklist Before Starting

- Related: #1140 (follow-up from #1139's review, both gaps described there)

### Test

- `tests/parallel/ulysses/test_wan_self_attn_padding_mask.py` (CPU-only): extended with coverage of the new `_trim_kv_tail_padding` helper (correct trim length, and a no-op when there's no mask).
- `tests/parallel/ulysses/test_wan_self_attn_ulysses.py` (new, multi-process): drives `SelfAttention.forward` directly with `ulysses_size=world_size`, gathers the SP-sharded output back across ranks, and checks it against a single-process non-SP reference computed from the same synced weights -- exercising the all-to-all fix end to end.
- `tests/parallel/ulysses/test_wan_ulysses_padding.py`: existing regression test, still passes.

All three run on 4xA10G (Modal): 4 passed / 1 passed / 1 passed, `ruff check` and `ruff format --check` clean.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved Wan self-attention accuracy when using Ulysses sequence parallelism, including synchronized and padded sequences.
  - Corrected handling of trailing padding for supported optimized attention backends.
  - Ensured attention behavior remains consistent with non-parallel reference results across distributed devices.

### CODE REVIEW

**Comment by OnePunchMonk** [comment]
### GPU test results (4xA10G, Modal)

Ran all three test files on a real 4-GPU box (image: CUDA 12.4.1, torch 2.5.1+cu124, transformers 5.9.0 -- matching the repo's pin -- pytest 7.4.4 per `pyproject.toml`'s dev pin).

**`tests/parallel/ulysses/test_wan_self_attn_padding_mask.py`** (CPU-only, 4 cases -- includes the 2 new ones for `_trim_kv_tail_padding`)
```
4 passed in 8.41s
```

**`tests/parallel/ulysses/test_wan_self_attn_ulysses.py`** (new, multi-process, `ulysses_size=4`)
```
Initializing parallel state: dp_size 1, dp_replicate_size 1, dp_shard_size 1, tp_size 1, pp_size 1, cp_size 1, ulysses_size 4, ep_size 1
Device mesh: DeviceMesh('cuda', [[0, 1, 2, 3]], mesh_dim_names=('dp_shard', 'ulysses'))
NCCL version 2.21.5+cuda12.4
1 passed in 20.37s
```
This drives `SelfAttention.forward` directly with Ulysses SP enabled across 4 ranks (no padding, so it isolates the all-to-all fix from the masking fix), gathers the SP-sharded output back across ranks, and compares it to a single-process non-SP reference computed from the same broadcast weights -- `torch.testing.assert_close(atol=2e-4, rtol=2e-3)`. Before the fix this fails at `init_parallel_state` isn't the issue -- the actual local-shard-only attention diverges from the full-sequence reference by much more than float noise; with the fix, the only difference is ~1e-5 GPU float32 reduction-order noise (1/4096 elements at 1.4e-5 before I loosened the tolerance from an initial 1e-5/1e-4 pass).

**`tests/parallel/ulysses/test_wan_ulysses_padding.py`** (existing regression, unmodified)
```
1 passed in 17.29s
```
Confirms the pre-existing issue #61 fix (sequence padding before the Ulysses slice) still round-trips losslessly with these changes.

**Lint**: `ruff check` and `ruff format --check` on the 3 changed files -- both clean.

Return code from the full run: `0`.

File: tests/parallel/ulysses/test_wan_self_attn_ulysses.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import torch
+import torch.distributed as c10d
+
+from veomni.utils.device import get_device_type, get_dist_comm_backend, get_torch_device
+
+
+if not c10d.is_available() or not c10d.is_backend_available(get_dist_comm_backend()):
+    print("c10d NCCL not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
```
[the remark this answers is not in the document]
Comment (OnePunchMonk) [reply tests/parallel/ulysses/test_wan_self_attn_ulysses.py]: Fixed in ca074836: switched the module-level `sys.exit(0)` guard to `pytest.skip(allow_module_level=True)`, so environments without the distributed backend now skip just this file instead of aborting the whole pytest session. Re-ran on 4xA10G (Modal) to confirm it still passes: `1 passed in 20.11s`.

**Review by Coach257 — APPROVED** [review]
